materials_io/adapters/base.py

```diff
@@ -62,3 +64,30 @@ class SerializeAdapter(BaseAdapter):
 
     def transform(self, metadata: dict, context=None) -> str:
         return json.dumps(metadata)
+
+
+class GreedySerializeAdapter(BaseAdapter):
```

Comment:
This is a good idea. We might want to promote the serialization of numpy arrays to another part of the library, but that's an issue for later.